man/data.table.Rd

```diff
@@ -96,7 +96,7 @@ data.table(\dots, keep.rownames=FALSE, check.names=FALSE, key=NULL, stringsAsFac
 
     See \href{../doc/datatable-intro.html}{\code{vignette("datatable-intro")}} and \code{example(data.table)}.}
 
-    \item{by}{ Column names are seen as if they are variables (as in \code{j} when \code{with=TRUE}). The \code{data.table} is then grouped by the \code{by} and \code{j} is evaluated within each group. The order of the rows within each group is preserved, as is the order of the groups. \code{by} accepts:
+    \item{by}{ Column names are seen as if they are variables (as in \code{j} when \code{with=TRUE}). The \code{data.table} is then grouped by the \code{by} and \code{j} is evaluated within each group. The order of the rows is preserved, regardless of any existing grouping. \code{by} accepts:
```

Comment:
sorry, what is an "existing grouping"?

could you elaborate what you find confusing about the existing wording?

Reply:
Got it, what the current wording is getting at is, in `DT[, x, by=y]`, (1) the values of `x` are in the same order they were found in the existing table, and (2) the values of `y` are in the same order they were found in the existing table.

"The order of rows is preserved" is not _exactly_ true, since ungrouped tables will be "shuffled" into `by` groups first.

Would "The order of groups is preserved, as is the order of rows within each group" be clearer?

Maybe

> The order of groups in the output will match what was found in the input; during computation within a group, the order of rows for a column will match what was found in the input.

Or

> {same wording as now} (See examples)

And make sure the examples illustrate well about row ordering using `by=`?

Reply:
I think that examples will be the best option to clarify this! Please give me some time, I'll find the some appropriate examples.

Reply:
Bump for interest. Still happy to add some more illustrative examples if you see fit.